jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -978,26 +979,35 @@ private void parsedHeader()
                         break;
 
                     case TRANSFER_ENCODING:
+                        _hasTransferEncoding = true;
+
                         if (_hasContentLength && complianceViolation(TRANSFER_ENCODING_WITH_CONTENT_LENGTH))
                             throw new BadMessageException(HttpStatus.BAD_REQUEST_400, "Transfer-Encoding and Content-Length");
 
-                        if (HttpHeaderValue.CHUNKED.is(_valueString))
-                        {
-                            _endOfContent = EndOfContent.CHUNKED_CONTENT;
-                            _contentLength = -1;
-                        }
-                        else
+                        // we encountered another Transfer-Encoding header, but chunked was already set
```

Comment:
Don't take out the short cut for checking `HttpHeaderValue.CHUNKED.is(_valueString)`.  That is the most common case and we really don't want to be doing CSV lists and iterators on every chunked header!

Reply:
done. fixed.